stl/inc/format

```diff
@@ -979,6 +979,34 @@ public:
             }
         }
     }
+    _NODISCARD _Decode_result<char> _Decode(const char* const _First, const char* const _Last, char32_t& _Val) const {
+        _STL_INTERNAL_CHECK(_First < _Last);
+        if (*_First == '\0') {
+            _Val = U'\0';
+            return {_First + 1, true};
+        }

```

Comment:
It appears that `_Decode_utf()` handles all of this. If so, I would recommend avoiding the repetition, with the following structure (warning: not actually compiled, this is just a sketch):

```cpp
if constexpr (_Is_ordinary_literal_encoding_utf8()) {
    return _Decode_utf(_First, _Last, _Val);
} else {
    // THIS LOGIC
    if constexpr (_Is_execution_charset_self_synchronizing()) {
        // MEOW
    } else {
        // WOOF
    }
}
```